test/category_runs_definitions.py

```diff
@@ -0,0 +1,117 @@
+import xarray
+
+
+class MockStream():
+    def __init__(self, metadata):
+        self.metadata = metadata
+        mock_image = [[1, 2], [3, 4]]
+        data_vars = {
+            'beamline_energy': ('time', [1, 2, 3, 4, 5]),
+            'i_zero': ('time', [-1, -2, -3, -4, -5]),
+            'ccd': (('time', 'dim_0', 'dim_1'), [mock_image, mock_image, mock_image, mock_image, mock_image])
+        }
+        self.dataset = xarray.Dataset(data_vars)
+
+    def to_dask(self):
+        return self.dataset.to_dask_dataframe()
+
+
+class MockRun():
+    def __init__(self, sample='', datafile='', num_images=3):
+        self.metadata = {
+            'start': {
+                'sample': '',
+                'data_file': ''
+            },
+            'stop': {
+                'num_events':
+                {'primary': 0}
+            }
+        }
+        self.metadata['stop']['num_events']['primary'] = num_images
+        self.metadata['start']['sample'] = sample
+        self.metadata['start']['data_file'] = datafile
+
+        metadata_with_descriptors = {'descriptors': [SAMPLE_DESCRIPTOR]}
+        metadata_with_descriptors.update(self.metadata)
+        self.primary = MockStream(metadata_with_descriptors)
+
+    def __getitem__(self, key):
+        if key == 'primary':
+            return self.primary
+        raise KeyError(f'Key: {key}, does not exist')
+
+    def __iter__(self):
+        yield 'primary'
+
+
+SAMPLE_DESCRIPTOR = {'data_keys': {'ai_0': {'dtype': 'number', 'shape': [], 'source': 'ai file'},
```

Comment:
Are all of these useful in your test? Can it be paired down, and make less-specific to this beamline?

Reply:
You're right. I can probably pair that down to only the values I'm actually using.

Reply:
Cool. I guess the reason is sticks out to me is that someone in the future will look at this and wonder what parts are significant to the test itself, which makes it a little harder to maintain.